apps/events/models.py

```diff
@@ -347,17 +346,21 @@ def _trigger(self):
         utc_now = timezone.now()
         self.last_triggered_at = utc_now
         self.total_triggers += 1
-        repetitions = self.params.get("repetitions", None)
-        if (repetitions is not None and self.total_triggers >= repetitions) or (
-            self.end_date and self.end_date <= timezone.now()
-        ):
+        if self._should_mark_complete():
             self.is_complete = True
         else:
             delta = relativedelta(**{self.params["time_period"]: self.params["frequency"]})
             self.next_trigger_date = utc_now + delta
 
         self.save()
 
+    def _should_mark_complete(self):
+        repetitions = self.params.get("repetitions", None) or 0
```

Comment:
Wondering if we shouldn't update the `repetitions` property to be something like

```python
    @cached_property
    def repetitions(self) -> int:
        return self.params["repetitions"] or 0
```

Reply:
We could do that. I'll update.